bindings/netstandard/ElectionGuard/ElectionGuard.ElectionSetup.Tests/KeyCeremony/TestElectionPolynomial.cs

```diff
@@ -0,0 +1,87 @@
+namespace ElectionGuard.ElectionSetup.Tests.KeyCeremony;
+
+public class TestElectionPolynomial
+{
+    [Test]
+    public void Test_ElectionPolynomial_Constructs_WithDegrees()
+    {
+        // Arrange
+        var degrees = 2;
+
+        // Act
+        var result = new ElectionPolynomial(degrees);
+
+        // Assert
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.Coefficients, Has.Count.EqualTo(degrees));
+    }
+
+    [Test]
+    public void Test_ElectionPolynomial_Constructs_WithSecretKey()
+    {
+        // Arrange
+        var degrees = 2;
+        var secretKey = Constants.TWO_MOD_Q;
+
+        // Act
+        var result = new ElectionPolynomial(degrees, secretKey);
+
+        // Assert
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.Coefficients, Has.Count.EqualTo(degrees));
+        Assert.That(result.Coefficients[0].Value, Is.EqualTo(secretKey));
+    }
+
+    [Test]
+    public void Test_ElectionPolynomial_Constructs_WithKeyPair()
+    {
+        // Arrange
+        var degrees = 3;
+        var secretKey = BigMath.RandQ();
+        var keyPair = new ElGamalKeyPair(secretKey);
+
+        // Act
+        var result = new ElectionPolynomial(degrees, keyPair);
+
+        // Assert
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.Coefficients, Has.Count.EqualTo(degrees));
+        Assert.That(result.Coefficients[0].Value, Is.EqualTo(keyPair.SecretKey));
+    }
+
+    // TODO: fix this test
+    [Ignore("throws bad alloc in elgamal keypair")]
+    [Test]
+    public void Test_ElectionPolynomial_ComputeCoordinate()
+    {
+        // Arrange
+        var coordinate = 1UL;
+        var degrees = 2;
+        var nonce = Constants.ONE_MOD_Q;
+        var seed = Constants.TWO_MOD_Q;
+        var polynomial = PolynomialExtensions.GeneratePolynomial(nonce, seed, degrees);
```

Comment:
the extensions are tossing up a bad alloc, which ill handle as part of the decryption work